### PR TITLE
feat(ITS): add chain name and token deploy salt and token sdk to ITS

### PR DESCRIPTION
**Pull request by ahramy**
[AXE-6809](https://axelarnetwork.atlassian.net/browse/AXE-6809)

[AXE-6809]: https://axelarnetwork.atlassian.net/browse/AXE-6809?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contracts/interchain-token-service/src/contract.rs [hunk 1]
```diff
@@ -146,17 +169,52 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
...
+        let _token_id = BytesN::<32>::from_array(env, &[0; 32]);
+
+        // TODO: _deployInterchainToken(deploySalt, currentChain, name, symbol, decimals, minterBytes, gasValue);
+
+        if initial_supply > 0 {
+            // TODO: mint, transferMintership
+        }
+    }
+
+    fn interchain_token_deploy_salt(env: &Env, deployer: Address, salt: BytesN<32>) -> BytesN<32> {
```
Comment (milapsheth) [inline contracts/interchain-token-service/src/contract.rs:204]: avoid changing the other interfaces in this PR to reduce conflicts with Attiss's PR. you can add this method and unit test it for now

File: contracts/interchain-token-service/src/contract.rs [hunk 2]
```diff
@@ -146,17 +169,52 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
...
+            .keccak256(
+                &(
+                    PREFIX_INTERCHAIN_TOKEN_SALT,
+                    chain_name_hash,
+                    deployer,
+                    salt,
+                )
+                    .to_xdr(env),
+            )
+            .into()
```
Comment (milapsheth) [inline contracts/interchain-token-service/src/contract.rs:217]: add a goldie test for it to ensure it doesn't change

Comment (ahramy) [reply contracts/interchain-token-service/src/contract.rs:217]: yup, added a goldie test

File: contracts/interchain-token-service/src/contract.rs [hunk 3]
```diff
@@ -155,6 +177,43 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
         _gas_token: Token,
     ) {
         todo!()
+
+        // caller.require_auth();
```
Comment (milapsheth) [inline contracts/interchain-token-service/src/contract.rs:181]: Avoiding adding commented code. Should be added properly in a separate PR

File: contracts/interchain-token-service/tests/test.rs [hunk 4]
```diff
@@ -175,3 +188,19 @@ fn transfer_ownership() {
...
+#[test]
+fn interchain_token_deploy_salt() {
+    let (env, client) = setup_env();
+
+    let deployer: Address = Address::from_str(
+        &env,
+        "CAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHK3M",
+    );
+
+    let salt = BytesN::<32>::from_array(&env, &[0; 32]);
```
Comment (milapsheth) [inline contracts/interchain-token-service/tests/test.rs:201]:
```suggestion
    let salt = BytesN::<32>::from_array(&env, &[1; 32]);
```
Maybe use a non zero value

**Review by milapsheth — APPROVED** [review]
